frontend/src/services/profile_handler_service.ts

```diff
@@ -0,0 +1,60 @@
+import { PersonProps, MockProfileBackendService } from "./mock_profile_backend";
+
+/** Error that occurs when Id does not exist */
+export class NonExistentProfileError extends Error {
+  constructor(readonly id: string) {
+    super(`Profile ${id} does not exist`);
+  }
+}
+
+/** Error that occurs if Id was unable to be updated */
+export class FailureToUpdateProfile extends Error {
+  constructor(readonly id: string) {
+    super(`Unable to update Profile ${id}`);
+  }
+}
+
+interface ProfileHandlerServiceInterface {
+  getPerson(id: string): Promise<string>;
+  updatePerson(PersonProps): Promise<boolean>;
+  deletePerson(id: string): Promise<boolean>;
+}
+
+/**
+ * Profile Handling service that manages list of user profiles using a mocked
+ * backend database.
+ */
+export class ProfileHandlerService implements ProfileHandlerServiceInterface {
+
+  /** Backend is responsible for holding all profile information */
+  constructor(private readonly backend: MockProfileBackendService) {}; 
```

Comment:
(the opposite of the last comment) This should be something like `BackendServiceInterface`, where for now we just happen to be passing in the mock version